tests/tensor_ops/test_iter.py

```diff
@@ -0,0 +1,45 @@
+import hypothesis.extra.numpy as hnp
+import pytest
+
+from mygrad.tensor_base import Tensor
+
+from ..wrappers.uber import backprop_test_factory, fwdprop_test_factory
+
+
+# Test https://github.com/rsokl/MyGrad/issues/210
+def test_0d_iter():
+    x = Tensor(3)
+    with pytest.raises(TypeError):
+        sum(x)
+
+
+def _sum(x, constant=False):
+    out = sum(x)
+    if isinstance(x, Tensor):
+        if not isinstance(out, Tensor):
+            # Hack to deal with summing over an empty tensor.
+            # `sum(Tensor([]))` returns 0, which is fine
+            out = Tensor(out)
+        out.constant = constant
+    return out
+
+
+@fwdprop_test_factory(
+    mygrad_func=_sum,
+    true_func=_sum,
+    num_arrays=1,
+    index_to_arr_shapes={0: hnp.array_shapes(min_dims=1, min_side=0)},
+)
+def test_builtin_sum_fwd():
+    pass
+
+
+@backprop_test_factory(
+    mygrad_func=_sum,
+    true_func=_sum,
+    num_arrays=1,
+    index_to_arr_shapes={0: hnp.array_shapes(min_dims=1, min_side=1)},
```

Comment:
Is the `min_side=1` is because `sum(mg.Tensor([]))` yields `0`?

Wouldn't it also exclude from the test valid cases like
```
>>> a = mg.arange(10*10*10).reshape(10,10,10)
>>> b = sum(a[:, slice(0, 1, -1), :]) # shape (10, 0, 10)
>>> b.backward()
```

(anytime when the 0-dim is anywhere except on the 0th axis)

Reply:
Yeah, this is a tricky case where `sum` returns `0` (an int, not a tensor), so we can't even invoke backprop.. I don't think there is any way to avoid this, since the tensor is actually an empty iterable and it is `sum` that returns 0.

Reply:
Oh dang... really nice catch. You are right about that!